drnim/drnim.nim

```diff
@@ -564,12 +568,299 @@ proc compatibleProps(graph: ModuleGraph; formal, actual: PType): bool {.nimcall.
       # we already know from the type system?
       result = frequires.isEmpty and fensures.isEmpty
 
+type
+  DrnimContext = ref object
+    graph: ModuleGraph
+    facts: seq[PNode]
+    o: Operators
+    hasUnstructedCf: int
```

Comment:
written but never read, and its name isn't clear, at least deserves a doc comment

Reply:
It'll be read. This PR is WIP.